code/obj/item/organ_holder.dm

```diff
@@ -57,6 +84,258 @@
 		if (src.donor && !ling) // so changers just get the datum and not a metric fuckton of organs
 			src.create_organs()
 
+	proc/build_region_buttons()
+
+		if (!src.chest)	//Can't do surgery without a chest to operate on
+			return null
+		src.contexts = list()
+
+		//begin by adding regions
+		var/datum/contextAction/surgery_region/ribs/ribs_action = new /datum/contextAction/surgery_region/ribs()
+		switch (src.ribs_stage)
+			if (0)
+				ribs_action.icon_background = "bg"
+			if (1)
+				ribs_action.icon_background = "yellowbg"
+			if (2)
+				ribs_action.icon_background = "greenbg"
```

Comment:
this should be abstracted, probably into the surgery_region contextAction parent